events/EventQueue.h

```diff
@@ -124,6 +124,21 @@ class EventQueue : private mbed::NonCopyable<EventQueue> {
      */
     void cancel(int id);
 
+    /** Query how much time is left for delayed event
+     *
+     *  If the event is delayed, this function can be used to query how much time
+     *  is left until the event is due to be dispatched.
+     *
+     *  This function is irq safe.
+     *
+     *  @param id       Unique id of the event
+     *
+     *  @return         Remaining time in milliseconds or
+     *                   0 if event is already due to be dispatched or
+     *                  -1 if invalid event id.
```

Comment:
Do you think we should actually return 0 for an invalid id?

Reply:
I was thinking that return value of invalid id should be somehow separated from 0 which is a valid value.
But actually the "invalid event id" seems to be a little problematic. Checking id 0 is clear but a random id is not. I don't think this check works properly (I just followed how cancel was implemented):
`if (e->id == id >> q->npw2)`
Just a quick test with id 99 gives following:
`e->id = 0, id = 99, q->npw2 = 10, id >> q->npw2 = 0`
so it is comparing 0 == 0 and therefore starts calculating diff from an invalid event.

Reply:
well, same with cancel, the user isn't allowed to pass an arbitrary int, only int ids given from a post function or the NULL event id 0.

For 99, the lookup in the array will point to arbitrary memory.